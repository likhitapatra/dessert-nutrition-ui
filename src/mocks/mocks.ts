import { GET_NUTRITION_DATA } from '../queries/queries';

export const mocks = [
  {
    request: {
      query: GET_NUTRITION_DATA,
      variables: {
        dessert: 'Oreo',
      },
    },
    result: {
      data: {
        nutritionData: { dessert: 'Oreo', nutritionInfo: {} },
      },
    },
  },
];
