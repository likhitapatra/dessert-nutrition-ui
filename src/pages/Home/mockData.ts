export const initialState = {
    nutritionData: [
        {
            dessert: 'Oreo',
            nutritionInfo: {
                calories: 437,
                fat: 18,
                carb: 63,
                protein: 4,
            },
        },
        {
            dessert: 'Nougat',
            nutritionInfo: {
                calories: 360,
                fat: 19,
                carb: 9,
                protein: 37,
            },
        },
    ]
}