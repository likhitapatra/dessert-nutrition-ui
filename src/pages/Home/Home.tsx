import React, { useEffect, useState } from 'react';
import Button from '../../components/Button/Button';
import Table from '../../components/Table/Table';
import Layout from '../../components/Layout/Layout';
import { useQuery } from '@apollo/client';
import { GET_NUTRITION_DATA } from '../../queries/queries';
import { initialState } from './mockData';

const Home = () => {
  const [state, setState]: any = useState({});

  const { loading, data } = useQuery(GET_NUTRITION_DATA, {
    fetchPolicy: 'cache-and-network',
  });

  useEffect(() => {
    if (data) {
      setState(data);
    }
  }, [data]);

  const handleReset = () => {
    setState(initialState);
  };

  return (
    <Layout>
      <div className="app-container">
        <div>
          <h3 className="f3 measure">Nutrition List</h3>
        </div>
        <div>
          <Button color="gray" onClick={handleReset}>
            Reset Data
          </Button>
        </div>
      </div>
      {loading ? 'Loading...' : <Table data={state} />}
    </Layout>
  );
};

export default Home;
