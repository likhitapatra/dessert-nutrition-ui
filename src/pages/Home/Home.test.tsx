import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import TestRenderer, { act } from 'react-test-renderer';
import Home from './Home';
import { mocks } from '../../mocks/mocks';
import { render } from '@testing-library/react';

describe('Home', () => {
  it('should render Home component', () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={mocks}>
        <Home />
      </MockedProvider>
    );

    const tree = component.toJSON();
    expect(tree?.type).toBe('div');
  });

  it('should match Home snapshot', async () => {
    let rendered;
    await act(async () => {
      rendered = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Home />
        </MockedProvider>
      );
    });
    expect(rendered).toMatchSnapshot();
  });
});
