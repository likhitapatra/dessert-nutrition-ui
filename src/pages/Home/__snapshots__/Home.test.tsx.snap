// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match Home snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pa4"
      >
        <div
          class="app-container"
        >
          <div>
            <h3
              class="f3 measure"
            >
              Nutrition List
            </h3>
          </div>
          <div>
            <button
              class="f6 link dim br1 ph3 pv2 mb2 dib white bg-dark-gray w-undefined"
            >
              Reset Data
            </button>
          </div>
        </div>
        Loading...
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pa4"
    >
      <div
        class="app-container"
      >
        <div>
          <h3
            class="f3 measure"
          >
            Nutrition List
          </h3>
        </div>
        <div>
          <button
            class="f6 link dim br1 ph3 pv2 mb2 dib white bg-dark-gray w-undefined"
          >
            Reset Data
          </button>
        </div>
      </div>
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
